### PR TITLE
UI Quickstart Guide - initial version

### PR DESCRIPTION
**Pull request by lzap**
This is the initial version of the UI Quickstart Guide - the landing page for all links within Foreman software itself.

The idea is to provide a consistent link experience, all HTML anchors which are linked from software are prefixed with "fmid" so we can easily identify those and never remove them. Since we have a HTML link validator available for authors and on our CI, this ensures all links in this document are also valid. When content gets moved or deleted, all we need to do is to update this guide as well.

I think it could be also useful to have screenshots for each linked page, but I do not have time for this. I think we can merge this without screenshots and ask our community to provide that later.

Open questions:

* Should we also add some text for each landing "term" / "chapter"? How much?
* Screenshots yes or no
* Should the links be bullets or a paragraph?
* Order of all "terms" / "chapters" - alphabetical?

### CODE REVIEW

**Comment by melcorr** [comment]
cc @maximiliankolb

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+If your network uses an HTTP Proxy, you can configure {Project} server to use an HTTP proxy for requests to EC2 Compute Resource.
+ifdef::katello,satellite[]
+Red Hat Content Delivery Network (CDN) or another content sources also use HTTP Proxy definition.
+endif::[]
+
+* {ProvisioningDocURL}Provisioning_Cloud_Instances_in_Amazon_EC2-Using-an-HTTP-Smart-Proxy[Using HTTP Proxy with Amazon EC2]
+* {InstallingProjectDocURL}configuring-satellite-with-an-http-proxy_foreman[Configuring an HTTP Proxy]
+
+// This is completely undocumented, or I haven't found it:
+To configure HTTP Proxy for other Compute Resources, go to *Administer* - *General* and set *HTTP(S) proxy*.
```
Comment (maximiliankolb) [inline guides/doc-User_Interface_Quickstart/master.adoc:123]:
```suggestion
To configure an HTTP(S) Proxy for other Compute Resources, go to *Administer > Settings* and select the *General* tab.
Set an *HTTP(S) proxy* for all outgoing HTTP connections from {Project}.
```

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+[[fmid-fact-values]]
+== Fact Values
+
+// TODO: This is missing from NG docs: https://theforeman.org/manuals/2.5/index.html#3.5.5FactsandtheENC
```
Comment (maximiliankolb) [inline guides/doc-User_Interface_Quickstart/master.adoc:128]: I've opened a new issue; IMHO it should be rather easy to migrate this from the Foreman Manual to the [puppet for configuration management guide](https://docs.theforeman.org/nightly/Managing_Hosts/index-foreman-el.html#using-puppet-for-configuration-management_managing-hosts). See https://github.com/theforeman/foreman-documentation/issues/579

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+A host is any Linux client that {Project} manages.
+For further information, visit {ManagingHostsDocURL}overview-of-hosts-in-satellite[Hosts Overview].
+
+* {ManagingHostsDocURL}Administering_Hosts[Administering Hosts]
+
+[[fmid-puppet-parametrized-classes]]
+== Parameterized Puppet Classes
+
+// TODO: Missing in NG docs: https://theforeman.org/manuals/2.5/index.html#4.2ManagingPuppet
```
Comment (maximiliankolb) [inline guides/doc-User_Interface_Quickstart/master.adoc:143]: See also https://github.com/theforeman/foreman-documentation/issues/579

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+It is possible to configure {Project} to enroll provisioned hosts to a FreeIPA realm group.
+
+* {AdministeringDocURL}sect-Administering-Configuring_External_Authentication-External_Authentication_for_Provisioned_Hosts[External Authentication for Provisioned Hosts]
+
+[[fmid-report-templates]]
+== Report Templates
+
+// TODO: Modularize
+You can use report templates to query Foreman data to obtain information about, for example, host status, registered hosts, applicable errata, applied errata, subscription details, and user activity.
```
Comment (maximiliankolb) [inline guides/doc-User_Interface_Quickstart/master.adoc:159]:
```suggestion
You can use report templates to query {Project} data to obtain information about, for example, host status, registered hosts, applicable errata, applied errata, subscription details, and user activity.
```

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+You can use the report templates that ship with Foreman or write your own custom report templates to suit your requirements.
```
Comment (maximiliankolb) [inline guides/doc-User_Interface_Quickstart/master.adoc:160]:
```suggestion
Use the report templates that ship with {Project} or write your own custom report templates that suit your requirements.
```

**Review by maximiliankolb — COMMENTED** [review]
I like the quickstart guide!

It looks like this PR has a couple of prerequisite that could/should be addressed in a different PR beforehand.

**Review by melcorr — CHANGES_REQUESTED** [review]
Looks fine but can you please provide an introductory text so that people know what this guide is supposed to be about?

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+* {ManagingHostsDocURL}using_report_templates[Using Report Templates to Monitor Hosts]
+* {ManagingHostsDocURL}appe-Red_Hat_Satellite-Managing_Hosts-Template_Writing_Reference[Template Writing Reference]
+* {ManagingHostsDocURL}Synchronizing_Templates_Repositories[Synchronizing Template Repositories]
+
+[[fmid-smart-proxies]]
+== Smart Proxies
```
[the remark this answers is not in the document]
Comment (lzap) [reply guides/doc-User_Interface_Quickstart/master.adoc:167]: Fixed.

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+The following documentation is available:
+
+* {ProvisioningDocURL}adding-kvm-connection_kvm-provisioning[KVM]
+* {ProvisioningDocURL}adding-rhv-connection_rhv-provisioning[oVirt]
+* {ProvisioningDocURL}Provisioning_Virtual_Machines_in_VMware_vSphere-Adding_a_VMware_vSphere_Connection_to_the_Satellite_Server[VMWare]
+* {ProvisioningDocURL}adding-kubevirt-connection_kubevirt-provisioning[KubeVirt]
+* {ProvisioningDocURL}adding-openstack-connection_openstack-provisioning[OpenStack]
+* {ProvisioningDocURL}Provisioning_Cloud_Instances_in_Amazon_EC2-Adding_a_Amazon_EC2_Connection_to_the_Satellite_Server[Amazon EC2]
+* {ProvisioningDocURL}adding-gce-connection_gce-provisioning[Google Compute Engine]
+* {ProvisioningDocURL}adding-azure-connection_azure-provisioning[Microsoft Azure]
```
[the remark this answers is not in the document]
Comment (lzap) [reply guides/doc-User_Interface_Quickstart/master.adoc:48]: Fixed.

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+TODO: We need some introduction for this topic, found nothing.
+
+For more information, see {ManagingHostsDocURL}customizing-the-registration-templates_managing-hosts[Customizing the Registration Templates] in _Managing Hosts_.
+
+[[fmid-compute-resources]]
+== Compute Resources
+
+A Compute Resource represents a virtual or cloud operating system instance.
+The following documentation is available:
```
[the remark this answers is not in the document]
Comment (lzap) [reply guides/doc-User_Interface_Quickstart/master.adoc:39]: Fixed.

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+* {ProvisioningDocURL}adding-openstack-details-to-a-compute-profile_openstack-provisioning[OpenStack]
+* {ProvisioningDocURL}Provisioning_Cloud_Instances_in_Amazon_EC2-Adding_Amazon_EC2_Details_to_a_Compute_Profile[Amazon EC2]
+* {ProvisioningDocURL}adding-gce-details-to-a-compute-profile_gce-provisioning[Google Compute Engine]
+* {ProvisioningDocURL}adding-azure-details-to-a-compute-profile_azure-provisioning[Microsoft Azure]
+
+For more information, see {ProvisioningDocURL}creating-compute-profiles_provisioning[Creating Compute Profiles] in _{ProjectName} Provisioning Guide_.
+
+[[fmid-puppet-config-groups]]
+== Puppet Config Groups
+
```
[the remark this answers is not in the document]
Comment (lzap) [reply guides/doc-User_Interface_Quickstart/master.adoc:68]: Isn't this exactly: * {ManagingHostsDocURL}using-puppet-for-configuration-management_managing-hosts[Using Puppet for Configuration Management]

File: guides/doc-User_Interface_Quickstart/master.adoc [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+//include::modules/con_puppet-introduction.adoc[leveloffset=+1]
+
+* {ManagingHostsDocURL}using-puppet-for-configuration-management_managing-hosts[Using Puppet for Configuration Management]
+* {ManagingHostsDocURL}puppet_guide_using_puppet_classes_managing-hosts[Puppet Classes and Config Groups]
+
+[[fmid-domains]]
+== Domains
+
+// TODO: Modularize this
+Foreman server defines domain names for each host on the network.
```
[the remark this answers is not in the document]
Comment (lzap) [reply guides/doc-User_Interface_Quickstart/master.adoc:79]: I will try modularized puppet section to see how it looks and let's see.

File: guides/doc-User_Interface_Quickstart/master.adoc (hunk 4 above)
Comment (lzap) [reply guides/doc-User_Interface_Quickstart/master.adoc:159]: Fixed. And couple of others.

File: guides/doc-User_Interface_Quickstart/master.adoc (hunk 5 above)
Comment (lzap) [reply guides/doc-User_Interface_Quickstart/master.adoc:160]: Fixed.

File: guides/doc-User_Interface_Quickstart/master.adoc (hunk 1 above)
Comment (lzap) [reply guides/doc-User_Interface_Quickstart/master.adoc:123]: Fixed.

**Comment by lzap** [comment]
I see no voices against this, so I will carry on. I am going to amend all the fixes for your comments.

For the order, I suggest the following:

* fmid-searching-and-bookmarking
* fmid-hosts
* fmid-smart-proxies
* fmid-domains
* fmid-subnets
* fmid-operating-systems
* fmid-provisioning-templates
* fmid-partition-tables
* fmid-realm
* fmid-puppet-config-groups
* fmid-puppet-parametrized-classes
* fmid-fact-values
* fmid-global-parameters
* fmid-compute-resources
* fmid-compute-profiles
* fmid-external-authentication
* fmid-report-templates
* fmid-http-proxies

I am going to write introduction and then do puppet which is fully modular to see how it looks like. Then we can take it from there. I think it is reasonable to modularize every introduction in this landing page as a separate PRs. Any help would be welcome!

File: guides/doc-User_Interface_Quickstart/master.adoc (hunk 9 above)
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/doc-User_Interface_Quickstart/master.adoc:68]: Ah, yes, I've missed the link in line 72.

**Review by maximiliankolb — APPROVED** [review]
Without looking at the generated output, LGTM.

**Comment by lzap** [comment]
This is no longer relevant I believe. Thanks for your time.

**Pull request closed without merging**